libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove.hpp

```diff
@@ -50,14 +252,18 @@ namespace hpx { namespace parallel { inline namespace v1 {
     namespace detail {
         /// \cond NOINTERNAL
 
-        // sequential remove_if with projection function
-        template <typename FwdIter, typename Pred, typename Proj>
-        FwdIter sequential_remove_if(
-            FwdIter first, FwdIter last, Pred&& pred, Proj&& proj)
+        template <typename Iter, typename Sent, typename UnaryPredicate,
+            typename Proj>
+        Iter sequential_remove_if(
+            Iter first, Sent last, UnaryPredicate p, Proj proj)
         {
-            return std::remove_if(first, last,
-                util::invoke_projected<Pred, Proj>(
-                    std::forward<Pred>(pred), std::forward<Proj>(proj)));
+            first = hpx::parallel::v1::detail::sequential_find_if(
+                first, last, p, proj);
+            if (first != last)
+                for (Iter i = first; ++i != last;)
+                    if (!p(*i))
```

Comment:
Two comments: 
- please use `hpx::util::invoke` for this
- should the result of `*i` be projected?

Reply:
Everything should be on its place now. I don't expect complains from CI either.